Documentation inconsistency: Minimum Python version requirement mismatch

### Description
I was following the installation_tutorial to build NEURON from source and encountered a confusing error during the CMake configuration step.

### Expected behavior vs. Actual behavior
In line 217 of documentation (docs/install/install_instructions.md), the minimum Python version required is stated as >=3.8:
```
Python >=3.8 (for Python interface)
```
However, when I ran CMake, it failed with the following error:

```
CMake Error at cmake/PythonHelper.cmake:198 (message):
  /path/to/python3 too old (3.8 < 3.10)
```

### Root cause
I found that in line 358 of the top-level CMakeLists.txt, the minimum Python version is actually set to 3.10:
```
set(NRN_MINIMUM_PYTHON_VERSION 3.10)
```
This creates a mismatch between the official documentation and the actual build system requirements.

### Proposed fix
Update the documentation (docs/install/install_instructions.md) to reflect the actual minimum Python version required, changing >=3.8 to >=3.10.

--- a/docs/install/install_instructions.md
+++ b/docs/install/install_instructions.md
@@ -214,7 +214,7 @@ In order to build NEURON from source, the following packages must be available:
 
 The following packages are optional (see build options):
 
-- Python >=3.8 (for Python interface)
+- Python >=3.10 (for Python interface)
 - Cython (for RXD)
 - MPI (for parallel)
 - X11 (Linux) or XQuartz (MacOS) (for GUI)

